stories/src/table/with-portal.jsx

```diff
@@ -0,0 +1,382 @@
+  // eslint-disable-next-line react/sort-comp
+  ref: ?HTMLElement
+
+  componentDidMount() {
+    const cellId: string = this.props.cellId;
+    if (!snapshotMap[cellId]) {
+      return;
+    }
+
+    if (!this.props.isDragging) {
+      // cleanup the map if it is not being used
+      delete snapshotMap[cellId];
+      return;
+    }
+
+    this.applySnapshot(snapshotMap[cellId]);
+  }
+
+  getSnapshotBeforeUpdate(prevProps: TableCellProps): ?TableCellSnapshot {
+    // we will be locking the dimensions of the dragging item on mount
+    if (this.props.isDragging) {
+      return null;
+    }
+
+    const isDragStarting: boolean = this.props.isDragOccurring && !prevProps.isDragOccurring;
+
+    if (!isDragStarting) {
+      return null;
+    }
+
+    return this.getSnapshot();
+  }
+
+  componentDidUpdate(
+    prevProps: TableCellProps,
+    prevState: mixed,
+    snapshot: ?TableCellSnapshot
+  ) {
+    const ref: ?HTMLElement = this.ref;
+    if (!ref) {
+      return;
+    }
+
+    if (snapshot) {
+      this.applySnapshot(snapshot);
+      return;
+    }
+
+    if (this.props.isDragOccurring) {
+      return;
+    }
+
+    // inline styles not applied
+    if (ref.style.width == null) {
+      return;
+    }
+
+    // no snapshot and drag is finished - clear the inline styles
+    ref.style.removeProperty('height');
+    ref.style.removeProperty('width');
+  }
+
+  componentWillUnmount() {
+    const snapshot: ?TableCellSnapshot = this.getSnapshot();
+    if (!snapshot) {
+      return;
+    }
+    snapshotMap[this.props.cellId] = snapshot;
+  }
+
+  getSnapshot = (): ?TableCellSnapshot => {
+    if (!this.ref) {
+      return null;
+    }
+
+    const { width, height } = this.ref.getBoundingClientRect();
+
+    const snapshot: TableCellSnapshot = {
+      width, height,
+    };
+
+    return snapshot;
+  }
+
+  applySnapshot = (snapshot: TableCellSnapshot) => {
+    const ref: ?HTMLElement = this.ref;
+
+    if (!ref) {
+      return;
+    }
+
+    if (ref.style.width === snapshot.width) {
+      return;
+    }
+
+    ref.style.width = `${snapshot.width}px`;
+    ref.style.height = `${snapshot.height}px`;
+  }
+
+  setRef = (ref: ?HTMLElement) => {
+    this.ref = ref;
+  }
+
+  render() {
+    return (
+      <Cell innerRef={this.setRef}>
+        {this.props.children}
+      </Cell>
+    );
+  }
+}
+
+type TableRowProps = {|
+  quote: Quote,
+  provided: DraggableProvided,
+  snapshot: DraggableStateSnapshot,
+  isDragOccurring: boolean,
+|}
+
+// Using a table as the portal so that we do not get react
+// warnings when mounting a tr element
+const table: HTMLElement = document.createElement('table');
+table.classList.add('my-super-cool-table-portal');
+Object.assign(table.style, {
+  margin: '0', padding: '0', border: '0',
+});
+const tbody: HTMLElement = document.createElement('tbody');
+table.appendChild(tbody);
+
+if (!document.body) {
+  throw new Error('document.body required for example');
+}
+document.body.appendChild(table);
```

Comment:
Will this add many tables because of HMR while in dev?

Reply:
For now I think it is fine